src/main/java/com/ibm/cloudant/kafka/connect/CloudantSourceTask.java

```diff
@@ -42,210 +40,218 @@
-
-					LOG.debug("Get continuous feed for lastSeq: " + latestSequenceNumber);
-
-					// the continuous changes feed
-					cantChangeResult = cantDB.changes()
-							.includeDocs(true)
-							.since(latestSequenceNumber)
-							.limit(batch_size)
-							//.continuousChanges() 
-								/*
-								 * => Removed because of performance (waiting time)
-								 * Requests Change notifications of feed type continuous. 
-								 * Feed notifications are accessed in an iterator style.
-								 * This method will connect to the changes feed; any configuration options applied after calling it will be ignored.	 
-								 */
-							.heartBeat(FEED_SLEEP_MILLISEC)
-							.getChanges();
-				}
-
-				LOG.debug("Got " + cantChangeResult.getResults().size() + " changes");
-				latestSequenceNumber = cantChangeResult.getLastSeq();
-
-				// process the results into the array to be returned
-				for (ListIterator<Row> it = cantChangeResult.getResults().listIterator(); it.hasNext(); ) {
-					Row row_ = it.next();
-					JsonObject doc = row_.getDoc();
-					Schema docSchema;
-					Object docValue;
-					if (generateStructSchema) {
-						docSchema = JsonStruct.jsonObjectToSchema(doc);
-						docValue = JsonStruct.jsonObjectToStruct(doc, docSchema);
-					} else {
-						docSchema = Schema.STRING_SCHEMA;
-						docValue = doc.toString();
-					}
-
-					// Emit the record to every topic configured
-					for (String topic : topics) {
-						SourceRecord sourceRecord = new SourceRecord(offsetKey(url), 
-								offsetValue(latestSequenceNumber), 
-								topic, // topics 
-								//Integer.valueOf(row_.getId())%3, // partition
-								Schema.STRING_SCHEMA, // key schema
-								row_.getId(), // key
-								docSchema, // value schema
-								docValue); // value
-						records.add(sourceRecord);
-					}
-				}
-
-				LOG.info("Return " + records.size() / topics.size() + " records with last offset " 
-				+ latestSequenceNumber);
-
-				cantChangeResult = null;
-				
-				_running.set(false);
-				return records;
-			}
-		}
 
-		// Only in case of shutdown
-		return null;
-	}
+    private static Logger LOG = Logger.getLogger(CloudantSourceTask.class);
+
+    private static long FEED_SLEEP_MILLISEC = 5000;
+    private static long SHUTDOWN_DELAY_MILLISEC = 10;
+    private static String DEFAULT_CLOUDANT_LAST_SEQ = "0";
+
+    private AtomicBoolean stop;
+    private AtomicBoolean _running;
+
+    private CloudantSourceTaskConfig config;
+
+    String url = null;
+    String userName = null;
+    String password = null;
+    List<String> topics = null;
+    int taskNumber;
+    boolean generateStructSchema = false;
+
+    private static String latestSequenceNumber = null;
+    private static int batch_size = 0;
+
+    private static CloudantClient cantClient = null;
+    private ChangesResult cantChangeResult = null;
+    private static Database cantDB = null;
+    private static String cantDBName = null;
+
+    public int counter = 0;
```

Comment:
This variable doesn't seem to be used in this class

Reply:
Fixed in 7c205d5719d12ff03091d5149405206e2ae498d5